contract-ui/components/primary-dashboard-button.tsx

```diff
@@ -30,68 +27,21 @@ export const PrimaryDashboardButton: React.FC<AddToDashboardCardProps> = ({
   const walletAddress = useAddress();
   const router = useRouter();
 
-  const oldRegistryContractList = useContractList(
-    chain?.chainId || 0,
-    chain ? getDashboardChainRpc(chain) : "",
-    walletAddress,
-  );
-
-  const newRegistryContractList = useMultiChainRegContractList(walletAddress);
-
-  const [addedState, setAddedState] = useState<"added" | "removed" | "none">(
-    "none",
-  );
+  const registry = useAllContractList(walletAddress);
 
   const { onSuccess: onAddSuccess, onError: onAddError } = useTxNotifications(
     "Successfully imported",
     "Failed to import",
   );
 
-  const onOldRegistry = useMemo(() => {
-    return (
-      oldRegistryContractList.isFetched &&
-      oldRegistryContractList.data?.find(
-        (c) => c.address === contractAddress,
-      ) &&
-      oldRegistryContractList.isSuccess &&
-      // contract can only be on the old registry if we haven't f'd with it
-      addedState === "none"
-    );
-  }, [
-    addedState,
-    contractAddress,
-    oldRegistryContractList.data,
-    oldRegistryContractList.isFetched,
-    oldRegistryContractList.isSuccess,
-  ]);
-
-  const onNewRegistry = useMemo(() => {
-    return (
-      (newRegistryContractList.isFetched &&
-        newRegistryContractList.data?.find(
-          (c) => c.address === contractAddress,
-        ) &&
-        newRegistryContractList.isSuccess) ||
-      // if we added it is on the new registry for sure
-      addedState === "added"
-    );
-  }, [
-    addedState,
-    contractAddress,
-    newRegistryContractList.data,
-    newRegistryContractList.isFetched,
-    newRegistryContractList.isSuccess,
-  ]);
-
-  const isAlreadyOnDashboard = useMemo(() => {
-    return (onOldRegistry || onNewRegistry) && addedState !== "removed";
-  }, [addedState, onNewRegistry, onOldRegistry]);
-
-  const statusUnknown = useMemo(() => {
-    return (
-      !oldRegistryContractList.isFetched || !newRegistryContractList.isFetched
-    );
-  }, [newRegistryContractList.isFetched, oldRegistryContractList.isFetched]);
+  const isOnRegistry =
```

Comment:
massive nitpick but wouldn't it be `is*In*Registry` - as in `existsinInRegistry` more so?

again doesn't matter, my brain just stumbled over reading `on` here

Reply:
_uses spanish native speaker as excuse_ but yes makes sense